File: 백준/Silver/13305. 주유소/주유소.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

	public static void main(String[] args) throws IOException {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int n = Integer.parseInt(br.readLine());

		long[] distances = new long[n - 1];
		long[] prices = new long[n];

		StringTokenizer st = new StringTokenizer(br.readLine());
		for (int i = 0; i < n - 1; i++) {
			distances[i] = Long.parseLong(st.nextToken());
		}

		st = new StringTokenizer(br.readLine());
		for (int i = 0; i < n; i++) {
			prices[i] = Long.parseLong(st.nextToken());
		}

		long minCost = 0;
		long minPrice = prices[0];

		for (int i = 1; i < n; i++) {
			minCost += minPrice * distances[i - 1];
			if (prices[i] < minPrice) {
				minPrice = prices[i];
			}
		}

		System.out.println(minCost);
	}
}

File: 백준/Silver/13305. 주유소/주유소.py
import sys

n = int(sys.stdin.readline().rstrip())

distances = list(map(int, sys.stdin.readline().rstrip().split()))
prices = list(map(int, sys.stdin.readline().rstrip().split()))

min_cost = 0
min_price = prices[0]

for i in range(1, n):
    min_cost += min_price * distances[i - 1]
    if prices[i] < min_price:
        min_price = prices[i]

print(min_cost)
